OloEngine/src/OloEngine/Renderer/Commands/CommandDispatch.cpp

```
@@ -99,34 +136,178 @@ namespace OloEngine
-        u32 SpotShadowTextureID = 0;
-        std::array<u32, UBOStructures::ShadowUBO::MAX_POINT_SHADOWS> PointShadowTextureIDs = { 0 };
+    // Helper: Bind legacy material textures (diffuse, specular).
+    static void BindLegacyTextures(const PODMaterialData& mat)
+    {
+        BindTrackedTexture(mat.diffuseMapID, ShaderBindingLayout::TEX_DIFFUSE, GL_TEXTURE_2D);
+        BindTrackedTexture(mat.specularMapID, ShaderBindingLayout::TEX_SPECULAR, GL_TEXTURE_2D);
+    }
 
-        // Snow accumulation depth texture (set per-frame)
-        u32 SnowDepthTextureID = 0;
+    static void UploadMaterialState(const PODMaterialData& mat, u16 materialDataIndex)
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'UploadMaterialState' of similar type are easily swapped by mistake

(bugprone-easily-swappable-parameters)